delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/indexing/ElasticSearchSink.scala

```diff
@@ -56,12 +58,19 @@ final class ElasticSearchSink private (
     if (bulk.nonEmpty) {
       client
         .bulk(bulk, refresh)
-        .redeemWith(
-          err =>
-            UIO
-              .delay(logger.error(s"Indexing in elasticsearch index ${index.value} failed", err))
-              .as(elements.map { _.failed(err) }),
-          _ => Task.pure(elements.map(_.void))
+        .map {
+          case NoErrors          => elements.map(_.void)
+          case SomeErrors(items) =>
+            elements.zip(Chunk.seq(items)).map {
+              case (element, BulkItemOutcome.Success)     => element.void
+              case (element, BulkItemOutcome.Error(json)) =>
+                element.failed(BulkUpdateException(json))
+            }
+        }
+        .onErrorHandleWith(err =>
+          UIO
+            .delay(logger.error(s"Indexing in elasticsearch index ${index.value} failed", err))
+            .as(elements.map(_.failed(err)))
```

Comment:
For the other errors (like a timeout), we don't want to mark the elems as failed but we want to let it crash.
The indexing will restart moments later, it will retry the same query